docs/reference/ilm/policy-definitions.asciidoc

```diff
@@ -0,0 +1,293 @@
+for shard filtering and/or replica counts. When updating number of replicas,
+configuring allocation rules is optional. When configuring allocation rules,
+setting number of replicas is optional. Although this action can be treated as
+two separate index settings updates, both can be configured at once.
+
+Read more about index replicas <<getting-started-shards-and-replicas,here>>.
+Read more about shard allocation filtering in
+the <<shard-allocation-filtering,Shard allocation filtering documentation>>.
+
+[[ilm-allocate-options]]
+.Allocate Options
+[options="header"]
+|======
+| Name                 | Required  | Default     | Description
+| `number_of_replicas` | no        | -           | The number of replicas to assign to the index
+| `include`            | no        | -           | assigns an index to nodes having at least _one_ of the values
+| `exclude`            | no        | -           | assigns an index to nodes having _none_ of the values
+| `require`            | no        | -           | assigns an index to nodes having _all_ of the values
+|======
+
+Although all configuration options are not required on their own. If `number_of_replicas` is not
+configured, then at least one of `include`, `exclude`, and `require` is required. An empty
+Allocate Action with no configuration is invalid.
+
+==== Example: Change number of replicas
+
+In this example, the index's number of replicas is changed to `2`, while allocation
+rules are unchanged.
+
+[source,js]
+--------------------------------------------------
+{
+  "allocate" : {
+    "number_of_replicas" : 2
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Assign index to node with specific "box_type" attribute
+
+This example assigns the index to nodes with `box_type` attribute of "hot" or "warm".
+
+[source,js]
+--------------------------------------------------
+{
+  "allocate" : {
+    "include" : {
+      "box_type": "hot,warm"
+    }
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Assign index to a specific node and update replica settings
+
+This example updates the index to have one replica per shard and be allocated
+to nodes with a `box_type` attribute of "cold".
+
+[source,js]
+--------------------------------------------------
+{
+  "allocate" : {
+    "number_of_replicas": 1,
+    "require" : {
+      "box_type": "cold"
+    }
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+[[ilm-delete-action]]
+=== Delete
+
+The Delete Action does just that, it deletes the index.
+
+This action does not have any options associated with it.
+
+[source,js]
+--------------------------------------------------
+{
+  "delete" : { }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+[[ilm-forcemerge-action]]
+=== Force Merge
+
+WARNING: This will make the index read-only (see: <<dynamic-index-settings,index.blocks.write>>)
```

Comment:
Also maybe this should not be a warning but a note? I worry that having it as a warning make it look like this is a bad thing to do when actually its expected in the normal workflow that this will be ok.

Reply:
++